magento-research/pwa-studio
# Fix makeUrl for Fastly

## Description

Fix `makeUrl` for Fastly image optimization.

## Related Issue

- #1038

## Motivation and Context

URLs from GraphQL are now absolute, and they include the media path logic we currently have in `makeUrl`. So we need to remove this logic and rewrite what remains to be compatible with absolute URLs. Also, we need to enable Fastly image optimization in certain environments.

## Verification

- Ensure category images are loading on the home page
- Ensure you can soft navigate (click) from home page to a category page
- Ensure product images are loading in the category gallery
- Ensure you can soft navigate from a category page to a product page
- Ensure product images are loading
- Ensure you can soft navigate to the home page via the header logo

## How Has This Been Tested?

- UAT with the above method, image optimization enabled
- UAT with the above method, image optimization disabled, Fastly enabled
- `yarn test`
